server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraJmapTestExtensionBuilder.java

```diff
@@ -0,0 +1,99 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import static org.apache.james.CassandraJamesServerMain.ALL_BUT_JMX_CASSANDRA_MODULE;
+
+import org.apache.james.mailbox.extractor.TextExtractor;
+import org.apache.james.mailbox.store.search.PDFTextExtractor;
+import org.apache.james.modules.TestESMetricReporterModule;
+import org.apache.james.modules.TestJMAPServerModule;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Module;
+import com.google.inject.util.Modules;
+
+public interface CassandraJmapTestExtensionBuilder {
+
+    class CoreModuleStage {
```

Comment:
the reason I didn't apply cascading lamdas is my stages will have custom value instead of the default one (in next commit in the future, when applying the extension to remaining integration tests - the code below describes the capability of my builder). Then I need more than one method for each stage rather than a single method FunctionalInterface instance.
E.g:
```java
Extension defaultSetup = builder().defaultExtension().defaultCoreModule().build();
Extension customExtensionSetup = builder().customExtension(myExtension).defaultCoreModule().build();
Extension customCoreModuleSetup = builder().defaultExtension().customCoreModule(myModule).build();
```
I couldn't find another elegant way to do this.

Reply:
Cascading lambda only apply for compulsory builder items.